algorithms/bfs/maze_search.py

```diff
@@ -0,0 +1,30 @@
+'''
+do BFS from (0,0) of the grid and get the minimum number of steps needed to get to the lower right column
+
+only step on the columns whose value is 1
+
+if there is no path, it returns -1
+'''
+
+def maze_search(grid):
+    dx = [0,0,-1,1]
+    dy = [-1,1,0,0]
+    n = len(grid)
+    m = len(grid[0])
+    q = [(0,0,0)]
+    visit = [[0]*m for _ in range(n)]
+    if grid[0][0] == 0:
+        return -1
+    visit[0][0] = 1
+    while q:
+        i, j, step = q.pop(0)
+        if i == n-1 and j == m-1:
```

Comment:
you are doing edge-case checking here (line 21) and line 26.
I guess we could consolidate it and keep only the bottom one?